src/test/gcp/cloudscheduler.spec.ts

```diff
@@ -0,0 +1,97 @@
+import * as _ from "lodash";
+import { expect } from "chai";
+import * as sinon from "sinon";
+import * as nock from "nock";
+import * as api from "../../api";
+
+import { cloudscheduler } from "../../gcp/";
+
+const VERSION = "v1beta1";
+
+const testJob = {
+  name: "projects/test-project/locations/us-east1/jobs/test",
+  schedule: "every 5 minutes",
+  timeZone: "America/Los_Angeles",
+  httpTarget: {
+    uri: "https://afakeone.come",
+    httpMethod: "POST",
+  },
+  retryConfig: {},
+};
+describe("cloudscheduler", () => {
+  let sandbox: sinon.SinonSandbox;
+
+  describe("createOrUpdateJob", () => {
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+      nock.cleanAll();
+    });
+
+    it("should create a job if none exists", async () => {
```

Comment:
in each of these tests blocks, add newlines so that there are three distinct blocks: align, act, and assert. Example below (comments may be omitted):

```typescript
      // ALIGN - setting up anything for the test to work.
      nock(api.cloudschedulerOrigin)
        .get(`/${VERSION}/${testJob.name}`)
        .reply(404, { context: { response: { statusCode: 404 } } });
      const mockJobResp = { schedule: "every 5 minutes" };
      nock(api.cloudschedulerOrigin)
        .post(`/${VERSION}/projects/test-project/locations/us-east1/jobs`)
        .reply(200, mockJobResp);

      // ACT - do the thing under test.
      const response = await cloudscheduler.createOrReplaceJob(testJob);

      // ASSERT - or in our case, expect something to validate the behavior.
      expect(response.body).to.deep.equal(mockJobResp);
```

Ping me for the internal documentation on this style.